### DISCUSSION TITLE
Sequential Order of Execution: Opening Apps vs Keystrokes

### DISCUSSION THREAD

**Discussion by alansmithee-dot** [ttscoff/bunch-docs#328] [Q&A]
I would like to open various applications one by one in the foreground and then use a global keystroke to throw the focused window to a specific space. 

Shortcuts for throwing focused windows is built into Amethyst (could also be created with other apps like Better Touch Tool or Keyboard Maestro). 

In Amethyst, the default shortcuts are Control-Option-Shift-# (space number e.g. 1, 2, 3, etc.)—this is the shortcut I’m using—so in Bunch that would be ^~$1, 2, 3, etc. 

So let's say I want to open Calendar, throw that to Space 1, then open Mail, and throw that to Space 7. 

I've created a bunch with the following text to be run sequentially:

Mail^
(pause 5)
{^~$1}

(pause 5)

Calendar^
(pause 5) 
{^~$7}

I append the caret so each app opens in the foreground and can be “thrown” to a space.  

I add pauses so each app can be fully loaded and thrown before the next app is loaded/thrown with the shortcut.

Bunch will open mail, pause 5 seconds, activate the keystroke to throw it to Space 1—then Bunch skips opening Calendar—pauses for 5 seconds, activates the keystroke to throw to Space 7, and as the final step, opens Calendar.

So this Bunch will open the first app, process all keystrokes that appear anywhere in the bunch file sequentially (skip opening apps in between), then go back through and open remaining apps sequentially.

I’ve scoured the Bunch manual and help forum to see if there’s any reference to what is happening here, but can’t figure out.

Is this intended?

I imagined sequential running would run each command one by one as written, but it seems keystrokes are prioritized over opening app commands (excluding the first app open).

Any explanation or alternate method to accomplish this would be greatly appreciated. Thank you!

**Comment by ttscoff** [comment]
Sequential execution is a bit of a weak point when it comes to mixing 
different types of apps, keystrokes, and commands. They don't execute 
like a checklist, and there's no intrinsic waiting for one to finish 
before executing the next. Your use of pauses in the right track, but 
you may have to increment each pause. See this topic on my forum:

https://forum.brettterpstra.com/t/using-moom-with-bunch-for-window-management/426/6

-Brett
